relays/ethereum/src/ethereum_deploy_contract.rs

```diff
@@ -59,18 +63,21 @@ pub fn run(params: EthereumDeployContractParams) {
 	let mut local_pool = futures::executor::LocalPool::new();
 
 	let result = local_pool.run_until(async move {
-		let eth_client = ethereum_client::client(params.eth);
-		let sub_client = substrate_client::client(params.sub);
+		let eth_client = EthereumRpcClient::new(params.eth);
+		let sub_client = SubstrateRpcClient::new(params.sub).await;
+
+		let sub_client = match sub_client {
```

Comment:
Doesn't `?` work?

Reply:
Derp, this match was the whole reason I went to the trouble of implementing `String` related traits for `RpcError`